### PR TITLE
Allow items passed to preCache to be Promises for URLs as well as just URLs

### PR DESCRIPTION
**Pull request by wibblymat**
R: @jeffposnick @addyosmani

### CODE REVIEW

File: lib/sw-toolbox.js [hunk 1]
```diff
@@ -32,7 +32,9 @@ self.addEventListener('install', function(event) {
   helpers.debug('preCache list: ' + (options.preCacheItems.join(', ') || '(none)'));
   event.waitUntil(
     helpers.openCache({cacheName: inactiveCache}).then(function(cache) {
-      return cache.addAll(options.preCacheItems);
+      return Promise.all(options.preCacheItems).then(function(items) {
```
Comment (jeffposnick) [inline lib/sw-toolbox.js]: I believe you can omit the anonymous function and just write `return Promise.all(options.preCacheItems).then(cache.addAll);`

Comment (addyosmani) [inline lib/sw-toolbox.js]: :+1:

Comment (wibblymat) [inline lib/sw-toolbox.js]: Needed a `bind`, but done.

**Comment by jeffposnick** [comment]
:+1:

**Pull request merged**